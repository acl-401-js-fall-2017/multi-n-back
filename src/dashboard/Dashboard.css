.dashboard {
    height: fit-content;
    overflow: auto;
    position: relative;
    z-index: -1;
}

.stats {
    height: 33em;
    border-radius: 5px;
    background: rgba(131, 199, 209, 0.7);    
    padding-top: 2.3em
}

.user-stats {
    position: absolute;
    top: 0; left: 0;
    margin: 2.5%;
    width: 30%;
    margin-right: 0;
    font-size: 50%;
}

.community-stats {
    position: inherit;
    font-size: 50%;
    left: 35%;
}

.community-leaderboards {
    width: 100%;
    height: 85%;
    display: flex;
    justify-content: space-around;
    box-sizing: border-box;
    padding: 0 1em;
}

.community-stats h2 {
    margin: 0 auto 1.5em auto;
    width: 15em;
    background-color: white;
    border-radius: 0.4em;
    box-shadow: 0 2px 4px rgb(131, 199, 209);
    border-bottom: solid #770d00 1px;
    padding: 0.3em 0.3em 0.4em 0.3em;
    letter-spacing: 0.1em
}

.community-leaderboard {
    box-sizing: border-box;
    width: 35%;
    padding: 1em;
    background-image: -webkit-gradient(linear, left top, left bottom, from(white), to(rgba(131, 199, 209, 0)));
    background-image: -webkit-linear-gradient(top, white, rgba(131, 199, 209, 0));
    background-image: -moz-linear-gradient(top, white, rgba(131, 199, 209, 0));
    background-image: linear-gradient(to bottom, white, rgba(131, 199, 209, 0));
    background-image: #d8eaec;
    border-top-left-radius: 0.5em;
    border-top-right-radius: 0.5em;
}

.community-leaderboard-header {
    width: 80%;
    margin: 0 auto 0.3em auto;
    border-bottom: 2px solid #770d00;
    padding-bottom: 0.5em;
    margin-bottom: 0.7em;
}




.community-highest-n,
.community-top-score{
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
}

.community-highest-n-leaderboard,
.community-top-score-leaderboard {
    display: flex;
    justify-content: center;
    height: 100%;
}

.community-leaderboard-of-highest-n,
.community-leaderboard-of-top-score {
    width: 32%;
    height: 100%;
    position: relative;
}

.community-leaderboard-of-highest-n:nth-child(-n + 2)::after,
.community-leaderboard-of-top-score:nth-child(-n + 2)::after {
    content: '';
    height: 100%;
    width: 1px;
    background-image: -webkit-gradient(linear, left top, left bottom, from(#254044), to(rgba(131, 199, 209, 0)));
    background-image: -webkit-linear-gradient(top, #254044, rgba(131, 199, 209, 0));
    background-image: -moz-linear-gradient(top, #254044, rgba(131, 199, 209, 0));
    background-image: linear-gradient(to bottom, #254044, rgba(131, 199, 209, 0));
    position: absolute;
    top: 0; right: -2px;
}




.community-most-recalled {
    width: 20%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
}

.community-most-recalled-leaderboard {
    height: 100%;
    margin-top: -0.3em; 
}




.community-leaderboard-difficulty-h5 {
    border-bottom: #254044 solid 1px;
    padding: 0.3em;
}

.community-leaderboard ul {
    list-style: none;
    margin: 0;
    padding: 0.15em;
    font-size: 70%;
    font-weight: lighter;
    height: 88%;
}

.community-leaderboard li {
    height: 1.8em;
}

.community-leaderboard span {
    display: inline-block;
    box-sizing: border-box;
}

.community-leaderboard-username {
    float: left;
    padding-left: 7%;
}

.community-leaderboard-data {
    float: right;
    padding-right: 7%;
}