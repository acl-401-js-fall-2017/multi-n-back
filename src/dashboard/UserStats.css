

.stats-page-left button {
    position: absolute;
    top: 0;
    left: 0; right: 0;
    width: 100%;
    height: 8%;
    border-radius: 0.4em;
    border: none;
    padding: 0.5em 2em 0.5em 2em;
    background: #568289;
    color: #d8eaec;
    letter-spacing: 0.1em;
    font-family: 'Gruppo', cursive;
    font-size: 150%;
    font-weight: bold;
}