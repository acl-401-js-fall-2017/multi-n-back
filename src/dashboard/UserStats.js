import React, { PureComponent } from 'react';
import { connect } from 'react-redux';

import './UserStats.css';

class UserStats extends PureComponent {
  
  render() {
    const { personal } = this.props;
    console.log('user stats personal??',personal);
    
    return personal ? (
      <article className="personal-leaderboard">
        <h4 className="personal-leaderboard-header">personal stats</h4>
        <div className="personal-leaderboard-stats">
          {Object.keys(personal).map(personalStat => (
            <section key={personalStat} className={`personal-leaderboard-stats-${personalStat} personal-leaderboard-of-all-stats`}>
              <h5 className="personal-leaderboard-stat-name-h5">{personalStat}</h5>
              <ul>
                <li>
                  <span className="personal-leaderboard-stat-data">{personal[personalStat]} </span>
                </li>
              </ul>
            </section>
          ))}
        </div>
      </article>
    ) : (
      <article className=""></article>
    );
  }
}

export default connect(
  state => ({}),
  null
)(UserStats);