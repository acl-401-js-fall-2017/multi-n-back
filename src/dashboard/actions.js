import * as actions from '../app/constants';
import * as stats from '../services/statsApi';


export function getCommunityStats() {
  return async dispatch => {

    const communityStats = await stats.getCommunityStats();
    console.log(communityStats);
    if(communityStats) dispatch({ 
      type: actions.GET_COMMUNITY_STATS,
      payload: communityStats
    });
  };
}

