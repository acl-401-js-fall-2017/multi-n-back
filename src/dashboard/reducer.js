import * as actions from '../app/constants';
import { combineReducers } from 'redux';

export const stats = combineReducers({ community, user });

function community(store = {}, { type, payload }) {
  switch (type) {
    case actions.GET_COMMUNITY_STATS:
      return payload;
    default:
      return store;
  }
}

function user(state = {}, { type, payload }) {
  switch(type) {
    case actions.GET_USER_STATS:
      return payload;
    default:
      return state;
  }
}