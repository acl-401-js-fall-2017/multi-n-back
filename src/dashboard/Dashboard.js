import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import UserStats from './UserStats';
import HighestN from './communityStats/HighestN';
import MostRecalled from './communityStats/MostRecalled';
import TopScore from './communityStats/TopScore';
import { getCommunityStats } from './actions';

import './Dashboard.css';

class Dashboard extends PureComponent {

  componentDidMount() { 
    this.props.getCommunityStats();
  }
  render() {
    const { user, stats } = this.props;
    const styleProps = user ? {
      width: '63%',
      margin: '2.5% 2.5% 2.5% 0'
    } : {
      zIndex: -1,
      width: '90%',
      left: '50%',
      transform: 'translate(-50%)'
    };
    return (
      <main className="dashboard">
        
        {user &&
          <UserStats/>
        }
        <section 
          className="stats community-stats"
          style={styleProps}
        >
          <h2>community stats</h2>
          <div className="community-leaderboards">
            <HighestN leaderboard={stats.highestNBackPerDifficulty}/>
            <MostRecalled leaderboard={stats.mostSequencesByUser}/>
            <TopScore leaderboard={stats.topScoresPerDifficulty}/>
          </div>
        </section>
      </main>
    );
  }
}

export default connect(
  state => ({
    user: state.user,
    stats: state.stats.community
  }),
  { getCommunityStats }
)(Dashboard);