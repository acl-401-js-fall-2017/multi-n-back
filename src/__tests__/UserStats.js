import React from 'react';
import { shallow } from 'enzyme';
import { UserStats } from '../dashboard/UserStats';

describe('UserStats', () => {
  it('shows UserStats component', () => {

    const wrapper = shallow(
      <UserStats getUserStats={() => {}}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});