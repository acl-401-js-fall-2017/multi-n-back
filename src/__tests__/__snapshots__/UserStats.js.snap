// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStats shows UserStats component 1`] = `<article />`;
